packages/common-utils/src/types.ts

```diff
@@ -125,7 +125,13 @@ export const SelectSQLStatementSchema = z.object({
     .array(
       z.object({
         name: z.string(),
-        sql: z.lazy(() => ChSqlSchema.or(ChartConfigSchema)),
+
+        // Need to specify either a sql or chartConfig instance. To avoid
+        // the schema falling into an any type, the fields are separate
+        // and listed as optional.
+        sql: ChSqlSchema.optional(),
+        chartConfig: z.lazy(() => ChartConfigSchema).optional(),
```

Comment:
curious the reason we use z.lazy here?

Reply:
It's to work around the recursive reference - `ChartConfig` is defined as an intersection with `SelectSQLStatementSchema` as one of the intersecting schemas and now `SelectSQLStatementSchema` can also be composed of other `ChartConfig` conforming objects. The only way that I've found to make this work in zod is through the use of `z.lazy`.